assembly/src/assembler/instruction/mod.rs

```diff
@@ -338,3 +346,15 @@ fn validate_param<I: Ord + Into<u64>>(value: I, min: I, max: I) -> Result<(), As
         Ok(())
     }
 }
+
+/// Casts input vector into Vec<Felt> and calls [env_ops::push] function
+fn cast_push_vec<T: Copy>(
+    imms: &[T],
+    span: &mut SpanBuilder,
+) -> Result<Option<CodeBlock>, AssemblyError>
+where
+    Felt: From<T>,
+{
+    let values: Vec<Felt> = imms.iter().map(|v| Felt::from(*v)).collect();
+    env_ops::push(&values, span)
+}
```

Comment:
it looks a bit odd that we need to allocate a vector just to transform it into a code block. Maybe we need a variant in `env_ops` that will instead take an iterator of `Felt` and perform create the code block from there? This way, we don't realloc